### PR TITLE
EOL zesty

### PR DESCRIPTION
**Pull request by mikaelarguedas**
All jobs are failing on Zesty singe the debs have been moved to old-releases (https://discourse.ros.org/t/suspension-of-debian-packaging-for-eol-ubuntu-zesty-17-04/3742)

An alternative would be to update the buildfarm to pull zesty debs from old-releases until the next Lunar sync and then EOL Zesty. As there is almost no changes since last sync I suggest to just EOL zesty right away

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Independent of when this is merged please make a PR to the `ros_buildfarm` repo to switch the apt url for  Zesty.

**Comment by mikaelarguedas** [comment]
> Independency of when this is merged please make a PR to the `ros_buildfarm` repo to switch the apt url for Zesty.

And then revert it ? or should all EOL distros be pointing to old-releases in the next release of `ros_buildfarm`?

**Comment by dirk-thomas** [comment]
> And then revert it ? or should all EOL distros be pointing to old-releases in the next release of `ros_buildfarm`?

It shouldn't be reverted. `ros_buildfarm` should continue to work for Zesty in case anyone would like to continue building for that platform. Since the apt repo url has changed that needs to be reflected in the buildfarm.

**Comment by mikaelarguedas** [comment]
Ok in that case I'll update for all distros that have been moved as the jobs will fail the same way with the current docker images / ros_buildfarm code (we'll need to update / rebuild the docker images used for building as well).

**Comment by mikaelarguedas** [comment]
https://github.com/ros-infrastructure/ros_buildfarm/pull/503 has been merged and https://github.com/ros-infrastructure/ros_buildfarm/pull/503#issuecomment-359564981 gives some context on potential future works and jobs that are expected to fail on EOL distros. Merging this... 

EDIT: I'll wait to merge this to see if we can get Zesty source jobs working before disabling them

**Comment by mikaelarguedas** [comment]
Holding this until last [Lunar Zesty sync](https://discourse.ros.org/t/preparing-for-lunar-sync-2018-01-22/3756/1) happens

**Comment by mikaelarguedas** [comment]
Zesty jobs have been [fixed](https://github.com/ros-infrastructure/ros_buildfarm/pull/504) and Lunar sync [happened](https://discourse.ros.org/t/new-packages-for-lunar-2018-01-24/3770), ready for review

**Review by dirk-thomas — APPROVED** [review]

**Review by nuclearsandwich — APPROVED** [review]
Please remember when [disabling/deleting jobs](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/doc/ongoing_operations.rst#deleting-all-views-related-to-a-ros-distribution) to batch requests in order to avoid http timeouts and the performance impacts they have.

**Comment by mikaelarguedas** [comment]
> Please remember when disabling/deleting jobs to batch requests in order to avoid http timeouts and the performance impacts they have.

:+1:

**Comment by mikaelarguedas** [comment]
Zesty jobs have been disabled using:

```groovy
import hudson.model.Cause

count = 0
for (p in Jenkins.instance.allItems) {
  if (
    (p.name.startsWith("Lsrc_uZ__") ||
    p.name.startsWith("Lbin_uZ64__") ||
    p.name.startsWith("Lrel_sync-packages-to-testing_zesty_amd64")) &&
    p.isBuildable())
  {
    println(p.name)
    count += 1
    p.disable()
  }
  if (count == 100) {
    break
  }
}
```

**Pull request merged**
